### PR TITLE
Add map.own

### LINKED ISSUE #143: map.own
## catchphrase
map own properties

## synopsis
```
any -> A; any -> B

map.own(mapper A=>B)(Object<A>) -> Object<B>
```

## description
**map.own** behaves as **map** on objects, the only difference being **map** iterates over all properties of an object whereas **map.own** iterates only on an object's direct properties not inherited from its prototype.

```javascript
const Person = function (name) {
  this.name = name
}

Person.prototype.greet = function () {
  console.log(`Hello, my name is ${this.name}`)
}

const david = new Person('david')

david.greet() // Hello, my name is david

david.a = 1
david.b = 2
david.c = 3

const square = number => number ** 2

map.own(square)(david) // { name: NaN, a: 1, b: 4, c: 9 }
```

## note
```javascript
const nativeHasOwnProperty = Object.prototype.hasOwnProperty

const objectHasOwnProperty = (object, property) => nativeHasOwnProperty.call(object, property)
```

### PR DESCRIPTION
**Pull request by LMulvey**
Closes #143 

This is my first contribution to this repo so by all means, let me know if there are any changes you need or if I've missed the mark. I tried to model it after existing methods as much as possible while matching the test case provided above. The tests could likely be expanded. Let me know if you'd like me to tackle that.

This is a super interesting library that I'll be using on my next personal project! Thanks for calling it out on Twitter to me.

### CODE REVIEW

File: map.js [hunk 1]
```diff
@@ -284,4 +286,11 @@ map.withIndex = mapper => function mappingWithIndex(value) {
   throw new TypeError(`${value} is not an Array`)
 }
```
Comment (richytong) [inline map.js:288]: The documentation at https://rubico.land/ starts right here in the comments. If you write comment documentation according to the following format, it all gets parsed into rubico.land via some scripts I set up.

**
 \* @name myFunc
 \*
 \* @synopsis
 \* \`\`\`coffeescript [specscript]
 \* myFunc(param1 any, param2 any) -> result any
 \* \`\`\`
 \*
 \* @description
 \* Hey, so below you can actually inline a code runner like this
 \* \`\`\`javascript [playground]
 \* console.log(
 \*   myFunc(1, 2),
 \* ) // something
 \* \`\`\`
 \*

**Review by richytong — CHANGES_REQUESTED** [review]
I'm impressed how you picked up the undocumented library conventions for naming/syntax. Thank you for this contribution, I've sent you some feedback via comments.

File: _internal/isObject.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * isObject(value any) -> boolean
+ *
+ * @description
+ * Determine whether a value is an object. Note that Arrays are also objects in JS.
+ */
+const isObject = value => {
+  if (value === null) {
+    return false
+  }
+  return (typeof value === 'object') || (typeof val === 'function')
```
[the remark this answers is not in the document]
Comment (LMulvey) [reply _internal/isObject.js:14]: Good point! That also surfaced a typo that wasn't allowing a `function` to be passed here (`val`->`value`).

File: map.js (hunk 1 above)
Comment (LMulvey) [reply map.js:288]: Fancy! I've updated the documentation here. Let me know if that requires any changes.

File: _internal/isObject.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ * @name isObject
+ *
+ * @synopsis
+ * isObject(value any) -> boolean
+ *
+ * @description
+ * Determine whether a value is an object. Note that Arrays are also objects in JS.
+ */
+const isObject = value => {
+  if (value === null) {
```
[the remark this answers is not in the document]
Comment (LMulvey) [reply _internal/isObject.js]: That makes sense 😄 . I've updated this condition, as well as the `typeof` tests below.

**Review by richytong — APPROVED** [review]
LGTM

**Pull request merged**